src/modules/ordersTable/pure/ReceiptModal/index.tsx

```diff
@@ -144,14 +156,27 @@ export function ReceiptModal({
               <OrderTypeField order={order} />
             </styledEl.Field>
 
-            <styledEl.Field>
-              {order.executionData.activityId && (
-                <>
-                  <FieldLabel label={order.executionData.activityTitle} />
-                  <IdField id={order.executionData.activityId} chainId={chainId} />
-                </>
-              )}
-            </styledEl.Field>
+            {/*TODO: add a link to explorer when it will support TWAP orders*/}
+            {!twapOrder && (
+              <styledEl.Field>
+                {order.executionData.activityId && (
+                  <>
+                    <FieldLabel label={order.executionData.activityTitle} />
+                    <IdField id={order.executionData.activityId} chainId={chainId} />
+                  </>
+                )}
+              </styledEl.Field>
+            )}
+            {twapOrder?.safeParams && (
```

Comment:
Can you make the safe params part more generic?

The goal is to show this also for non-TWAP but still Safe orders

Reply:
Make sense. Refactored:
https://github.com/cowprotocol/cowswap/pull/2652/commits/4b8849976697861431e828f90b806447f6fe1daf